pandas/tests/libs/test_lib.py

```diff
@@ -248,6 +248,13 @@ def test_is_range_indexer(self, dtype):
         left = np.arange(0, 100, dtype=dtype)
         assert lib.is_range_indexer(left, 100)
 
+    @pytest.mark.parametrize("dtype", ["int64", "int32"])
+    def test_is_range_indexer_big_n(self, dtype):
+        # GH53616
+        left = np.arange(0, 100, dtype=dtype)
+
+        assert not lib.is_range_indexer(left, 2**31)
```

Comment:
```suggestion
        assert not lib.is_range_indexer(left, np.iinfo(np.int32).max)
```

To fix the failing 32 bit build

Reply:
Thanks for the catch. I can't change the size here unfortunately (need it to be ``2**31 == np.iinfo(np.int32).max + 1`` to trigger the OverflowError previously).

Gonna skip on 32-bit instead, since the issue can't happen there.
(Max array len should be less than int32 max on 32-bit platforms).